tests/mason-lspconfig/api/api_spec.lua

```diff
@@ -0,0 +1,55 @@
+local mason_lspconfig = require "mason-lspconfig"
+local _ = require "mason-core.functional"
+
+describe("mason-lspconfig API", function()
+    it("should return all available servers", function()
+        local server_mappings = require "mason-lspconfig.mappings.server"
+        local available_servers = mason_lspconfig.get_available_servers()
+        assert.equal(#vim.tbl_keys(server_mappings.package_to_lspconfig), #available_servers)
+    end)
+
+    it("should return all available servers for given filetype", function()
+        assert.same(
+            { "terraformls", "tflint" },
+            _.sort_by(
+                _.identity,
+                mason_lspconfig.get_available_servers {
+                    filetype = "terraform",
+                }
+            )
+        )
+    end)
+
+    it("should return all available servers for given filetypes", function()
+        assert.same(
+            { "lemminx", "taplo" },
+            _.sort_by(
+                _.identity,
+                mason_lspconfig.get_available_servers {
+                    filetype = { "xml", "xsd", "xsl", "toml" },
+                }
+            )
+        )
+    end)
+
+    it("should return no servers if filetype predicate has no matches", function()
+        assert.same(
+            {},
+            mason_lspconfig.get_available_servers {
+                filetype = { "thisfiletypesimplydoesntexist" },
+            }
+        )
+    end)
+
+    it("should handle servers without a matching registry", function()
+        local filetype_mappings = require "mason-lspconfig.mappings.filetype"
+        filetype_mappings.dummylang = { "dummylsp", "dummy2lsp" }
+        assert.same(
+            {},
+            mason_lspconfig.get_available_servers {
+                filetype = "dummylang",
+            }
+        )
```

Comment:
Hmm this doesn't seem like intended behaviour to me. The test fixtures are super hacky, I just shoehorned them in for some specific test cases.

I'd expect this to return `{ "dummylsp", "dummy2lsp" }`?  Mason doesn't support registering custom packages yet, so this is in many ways undefined behaviour, there's no right or wrong

Reply:
> I'd expect this to return `{ "dummylsp", "dummy2lsp" }`? Mason doesn't support registering custom packages yet, so this is in many ways undefined behaviour, there's no right or wrong

yeah, that's what I had expected as well. I'll add another item to the TODO list for traceability in later PRs.

Reply:
@williamboman, anything else before merging this?